tests/test_users.py

```diff
@@ -0,0 +1,198 @@
+import os
+import json
+
+import pytest
+
+os.environ['DATABASE'] = 'users_test.db'
+
+import create_data_base
+
+from app import app
+
+
+@pytest.fixture
+def client():
+    app.config['TESTING'] = True
+
+    with app.test_client() as client:
+        yield client
+
+
+@pytest.fixture
+def user(client):
+    data = {
+        'name': 'User',
+        'last_name': 'Test',
+        'email': 'TEST@test.com',
+        'date': 'none',
+        'password': 'test-test'
+    }
+    response = client.post('/users', data=json.dumps(data))
+    user = json.loads(response.data)
+    user.update(data)
+
+    yield user
+
+    client.delete('/users/{}'.format(user['id']))
+
+
+###################### post tests ##########################
+
+def test_users_post(client):
+    data = {
+        'name': 'pablo',
+        'last_name': 'diaz_ogni',
+        'email': 'pbldo@perro.com',
+        'date': 'none',
+        'password': 'ehehehehhe'
+    }
+
+    response = client.post('/users', data=json.dumps(data))
+    user = json.loads(response.data)
+
+    assert response.status_code == 200
+    assert 'id' in user 
+    client.delete('/users/{}'.format(user['id']))
+
+
+def test_users_post_400_missing_field(client):
+    #'name': 'pablo', FALTARÍA EL NOMBRE
+    data = {
+        'last_name': 'diaz_ogni',
+        'email': 'pbldo@perro.com',
+        'date': 'none',
+        'password': 'ehehehehhe'
+    }
+
+    response = client.post('/users', data=json.dumps(data))
+    print(json.loads(response.data))
+    assert response.status_code == 400
```

Comment:
Some awesome feature of `pytest` is that you can write a single test and create a list of different values to test, so it will run the test for each value but reusing the same code. 

https://docs.pytest.org/en/stable/parametrize.html

Example:
```python
@pytest.mark.parametrize(
    "field",  # This is the name that will use the test case to receive each argument on each run
    ["name", "last_name", "email"]
)
def test_users_post_400_missing_field(client, field):
    # Set ALL the fields
    data = {
        'name':..,
        'email':...,
        
    }
    # Pop de field you want to test on each test case:
    data.pop(field)
    
    response = client.post(...
    assert response.status_code == 400
```

The result will be that you will have 3 tests:

test_users_post_400_missing_field_name
test_users_post_400_missing_field_last_name
test_users_post_400_missing_field_email

Reply:
Remember to import pytest ;)